### DISCUSSION TITLE
ue4 的kbe 插件编译报错 error C2653: 'EntityDef': is not a class or namespace

### DISCUSSION THREAD

**Discussion by MacroGu** [kbengine/kbengine#1363] [General]
报错信息如下：
error C2653: 'EntityDef': is not a class or namespace name
生成的插件代码，是在 EntityCallExRoleBase.cpp 这个文件中， 对应的这段代码报错：
`void EntityCellEntityCall_ExRoleBase::AnimUpdate(const ANIM_INFO& arg1)
{
Bundle* pBundleRet = newCall("AnimUpdate", 0);
if(!pBundleRet)
return;

((DATATYPE_ANIM_INFO*)EntityDef::id2datatypes[29])->addToStreamEx(*pBundleRet, arg1);
sendCall(NULL);
}`
这段代码在我给实体新增了一些属性和方法后， 重新生成插件编译就会报错， 请问，是我实体定义的问题吗？

**Comment by MacroGu** [comment]
我在 EntityCallExRoleBase.cpp 文件中添加了 
#include "EntityDef.h"
这行代码， 就可以编译成功了， 是不是生成插件的时候，没有引入？

**Reply by kebiao** [reply]
可能是ue4版本变动造成

**Reply by MacroGu** [reply]
添加这个 头文件后， 编译就通过了， 需要在UE4 生成的脚本代码里面，自动加上生成引入这一句：#include "EntityDef.h"
 的代码吗？ 如果需要请通过我的PR，谢谢
https://github.com/kbengine/kbengine/issues/1361
